lib/ansible/modules/cloud/vmware/vmware_dvswitch.py

```diff
@@ -1,76 +1,88 @@
 #!/usr/bin/python
 # -*- coding: utf-8 -*-
-
+#
 # Copyright: (c) 2015, Joseph Callen <jcallen () csc.com>
+# Copyright: (c) 2018, Abhijeet Kasurde <akasurde@redhat.com>
+# Copyright: (c) 2018, Ansible Project
+#
 # GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
 
 from __future__ import absolute_import, division, print_function
 __metaclass__ = type
 
-ANSIBLE_METADATA = {'metadata_version': '1.1',
-                    'status': ['preview'],
-                    'supported_by': 'community'}
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
 
 DOCUMENTATION = '''
 ---
 module: vmware_dvswitch
 short_description: Create or remove a distributed vSwitch
 description:
-    - Create or remove a distributed vSwitch
+    - This module can be used to create, remove a distributed vSwitch.
 version_added: 2.0
 author:
 - Joseph Callen (@jcpowermac)
+- Abhijeet Kasurde (@Akasurde)
 notes:
     - Tested on vSphere 6.5
 requirements:
     - "python >= 2.6"
     - PyVmomi
 options:
     datacenter_name:
-        description:
-            - The name of the datacenter that will contain the dvSwitch
-        required: True
+      description:
+      - The name of the datacenter that will contain the distributed vSwitch.
+      required: True
     switch_name:
-        description:
-            - The name of the switch to create or remove
-        required: True
+      description:
+      - The name of the distribute vSwitch to create or remove.
+      required: True
     switch_version:
-        description:
-            - The version of the switch to create. Can be 6.5.0, 6.0.0, 5.5.0, 5.1.0, 5.0.0 with a vcenter running vSphere 6.5
-            - Needed if you have a vcenter version > ESXi version to join DVS. If not specified version=version of vcenter
-        required: False
-        version_added: 2.5
+      description:
+      - The version of the distributed vSwitch to create.
+      - Can be 6.5.0, 6.0.0, 5.5.0, 5.1.0, 5.0.0 with a vCenter running vSphere 6.5.
+      - Needed if you have a vcenter version > ESXi version to join DVS. If not specified version=version of vCenter.
+      - Required only if C(state) is set to C(present).
+      version_added: 2.5
+      choices: ['5.0.0', '5.1.0', '5.5.0', '6.0.0', '6.5.0']
```

Comment:
6.6.0 should be added for the latest version (available in vSphere 6.7), including updating the status of vSphere 6.5 to vSphere 6.7 above

Reply:
There should be a comment as well that vSphere 6.7 doesn't support 5.0.0, 5.1.0, and 5.5.0 anymore. Or we should check that in the code and throw an error...

Reply:
I don't have 6.7 setup to test this and also this is very old patch. :)

Reply:
I changed it in my version of your PR. I just need to know how to share. See my comment below. I added the update feature in the meantime as well...